### PR TITLE
fixme: adding debounce util + removing underscore as dependency

### PR DESCRIPTION
**Pull request by mannynotfound**
tackling some TODO's while reading the source code. This seemed like a good one. No need to require all of underscore for one func and makes it more flexible for people who dont want underscore (lodash stans for example)

New debounce util is just underscore's version in ES6 + conforming to eslint rules.

### CODE REVIEW

File: src/utils/debounce.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  let timeout;
+  let args;
+  let context;
+  let timestamp;
+  let result;
+
+  const later = () => {
+    const last = now() - timestamp;
+
+    if (last < wait && last >= 0) {
```
Comment (rolyatmax) [inline src/utils/debounce.js:19]: This must be covering an interesting edge case. Maybe there are scenarios where `later()` gets called before `wait` has elapsed?

**Comment by rolyatmax** [comment]
This all looks good to me! The only thing I might add in is a comment next to `debounce` indicating that the implementation was based on `underscore`'s in case there are future questions as to why it's covering edge cases and using an `immediate` arg, etc.

**Comment by rolyatmax** [comment]
Thanks for doing this @mannynotfound! Gotta love decreasing the size of the bundle!

**Comment by mannynotfound** [comment]
@rolyatmax added the comment here https://github.com/rolyatmax/redux-resource-manager/pull/11/commits/52edef6e9e071f746fddc13b9d47e9a32cc0121d

**Comment by rolyatmax** [comment]
❤️  Thanks! Looks great!

**Pull request merged**